### PR TITLE
Make pprint/pformat tolerant of non-UTF-8 bytes data

### PR DESCRIPTION
**Pull request by taldcroft**
Currently if a bytestring column has content that is not UTF-8 compliant you cannot get the repr of that column:
```
In [3]: t = Table([[b'Z\xf0']])

In [4]: t
Out[4]: ---------------------------------------------------------------------------
UnicodeDecodeError                        Traceback (most recent call last)
...
UnicodeDecodeError: 'utf8' codec can't decode byte 0xf0 in position 1: unexpected end of data
```
This fixes that by decoding with `errors='replace'`.

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -323,6 +323,9 @@ Bug Fixes
 
 - ``astropy.tests``
 
+  - Fix problem with Table pprint/pformat raising an exception
```
Comment (bsipocz) [inline CHANGES.rst]: isn't it ``astropy.table`` rather than ``astropy.tests``?

Comment (pllim) [inline CHANGES.rst]: Yes, should be table.

Comment (taldcroft) [inline CHANGES.rst]: Do'oh!  Fixed.

**Comment by mhvk** [comment]
Looks OK in general, though should it be `namereplace_errors` for Python >=3.5? (we'll be there soon enough...). Also, how does this interact with #5700 - do we still get bytes there? With that in place, should any bytes just be typeset as `b'...'`?

I guess my bottom line: definitely for 1.3.3, but let's be sure we get it right for 2.0.

**Comment by taldcroft** [comment]
It looks to me like `namereplace_errors()` is only for encoding.

**Comment by bsipocz** [comment]
Based on the discussion above, this seemed to be agreed on to be good to go into 1.3.3. I'm cutting the release now, thus merging.

**Pull request merged**